packages/account/src/BiconomySmartAccount.ts

```diff
@@ -464,4 +473,72 @@ export class BiconomySmartAccount extends SmartAccount implements IBiconomySmart
   async getAllSupportedChains(): Promise<SupportedChainsResponse> {
     return this.nodeClient.getAllSupportedChains();
   }
+
+  async getUpdateImplementationData(newImplementationAddress?: string): Promise<Transaction> {
+    // V2 address or latest implementation if possible to jump from V1 -> Vn without upgrading to V2
+    // If needed we can fetch this from backend config
+
+    Logger.log("Recommended implementation address to upgrade to", BICONOMY_IMPLEMENTATION_ADDRESSES_BY_VERSION.V2_0_0);
+
+    const latestImplementationAddress = newImplementationAddress ?? DEFAULT_BICONOMY_IMPLEMENTATION_ADDRESS; // BICONOMY_IMPLEMENTATION_ADDRESSES_BY_VERSION.V2_0_0 // 2.0
+    Logger.log("Requested implementation address to upgrade to", latestImplementationAddress);
+
+    // by querying the proxy contract
+    // TODO: add isDeployed checks before reading below
+    const currentImplementationAddress = await this.proxy.getImplementation();
+    Logger.log("Current implementation address for this Smart Account", currentImplementationAddress);
+
+    if (ethers.utils.getAddress(currentImplementationAddress) !== ethers.utils.getAddress(latestImplementationAddress)) {
+      const impInterface = new ethers.utils.Interface(["function updateImplementation(address _implementation)"]);
+      const encodedData = impInterface.encodeFunctionData("updateImplementation", [latestImplementationAddress]);
+      return { to: this.address, value: BigNumber.from(0), data: encodedData };
+    } else {
+      // Could throw error instead
+      // throw new Error("Not eligible for upgrade");
```

Comment:
Don't think we will need this, below we are already throwing error for 0x. Can remove this comment

Reply:
> below we are already throwing error for 0x

where?

Reply:
ok I found it.

Reply:
updated